helix-term/src/ui/popup.rs

```diff
@@ -198,20 +198,29 @@ impl<T: Component> Popup<T> {
             && y < self.area.bottom();
 
         if !mouse_is_within_popup {
-            return EventResult::Ignored(None);
+            return EventResult::Ignored(Some(self.close_cb()));
```

Comment:
not quite sure how I feel about closing popups without giving their contents a chance to cleanup.

I think we should emulate a `c-c` press here so the popup gets a chance to cleanup 

```suggestion
            let _ = self.contents.handle_event(ctrl!(c), cx)
            return EventResult::Ignored(Some(self.close_cb()));
```